tests/models/test_ModelUtils.py

```diff
@@ -6,22 +6,113 @@
 import pytest
 import torch
 import torch.nn as nn
+from packaging import version
 from pytest_mock import MockerFixture
 from torch import Tensor
 from torch.nn import Identity, Parameter
 
 from lightly.models import utils
 from lightly.models.utils import (
+    _mask_reduce,
+    _mask_reduce_batched,
     _no_grad_trunc_normal,
     activate_requires_grad,
     batch_shuffle,
     batch_unshuffle,
     deactivate_requires_grad,
+    masked_pooling,
     nearest_neighbors,
     normalize_weight,
     update_momentum,
 )
 
+scatter_min_torch_version = version.parse("1.12.0")
+torch_version = version.parse(torch.__version__)
```

Comment:
Instead of depending on `packaging` we usually check if the feature is available:

```suggestion 
is_scatter_reduce_available = hasattr(Tensor, "scatter_reduce_")
```